src/lib/database/schemas/Version1Schema.ts

```diff
@@ -0,0 +1,98 @@
+import { Transaction } from "dexie"
+import {z} from "zod"
+
+
+export const DbPositionZod = z.enum([
+  "Stand",
+  "Exposure",
+  "Korean",
+  "Sofa",
+  "Chest",
+  "Back",
+  "Belly",
+  "Shoulder",
+  "Hang",
+  "Nevermind",
+  "Dropknee",
+  "Double Drop Knee",
+  "Inward Drop Knee",
+  "Leash",
+  "Kneehang",
+  "Soup",
+  "Sit",
+  "Rocket",
+  "Frodo",
+  "Buddha",
+  "Yisus",
+  "Banana",
+])
+
+
+export const DbStickableStatusZod = z.enum(["official", "userDefined", "archived"])
+
+
+// [id, trickStatus] OR [id, comboStatus]
+const DbReferenceZod = z.tuple([z.number().int(), DbStickableStatusZod])
+
+
+const DbVideoZod = z.object({
+  link: z.string().nonempty().url(),
+  startTime: z.number().min(0).optional(),
+  endTime: z.number().min(0).optional()
+})
+
+
+const CommonDefinition = {
+  id: z.number().int(),
+  technicalName: z.string().nonempty(),
+  alias: z.string().nonempty().optional(),
+  establishedBy: z.string().nonempty().optional(),
+  yearEstablished: z.number().int().positive().optional(),
+  description: z.string().nonempty().optional(),
+  tips: z.array(z.string().nonempty()).optional(),
+  dateAddedEpoch: z.number().int().min(0),
+  videos: z.array(DbVideoZod).optional()
+} as const
+
+
+export const DbTricksTableZod = z.object({
+  ...CommonDefinition,
+  trickStatus: DbStickableStatusZod,
+  startPosition: DbPositionZod,
+  endPosition: DbPositionZod,
+  difficultyLevel: z.number().int().min(0),
+  recommendedPrerequisites: z.array(DbReferenceZod).optional(),
+  variationOf: z.array(DbReferenceZod).optional(),
+  showInSearchQueries: z.boolean(),
+})
+
+export const DbCombosTableZod = z.object({
+  ...CommonDefinition,
```

Comment:
I'm not very familiar with some of the packages and syntax used here but does this infer, that combos have both a "technicalName" and an "alias" in their DB schema? If so

1. Combos should not have a technical name
2. The name of a combo should be called "name" not "alias"

Reply:
yep. That's whats happening here. Because the object is defined "as const" you can spread the definition and TS does pick it up correctly.